src/services/github.ts

```diff
@@ -314,16 +344,14 @@ export async function githubUpload(context: ContextType<typeof AppContext>, base
         existingFigures = [];
     }
 
-    const figurePaths: string[] = existingFigures.map((f: { path: string; }) => f.path);
-    let i = 0;
-    let proposedName, proposedPath;
-    do {
-        proposedName = name.substring(0, name.lastIndexOf(".")) + ( i ? "_" + (i+1) : "") + name.substring(name.lastIndexOf("."));
-        proposedPath = figurePath + "/" + proposedName;
-        i++;
-    } while(figurePaths.includes(proposedPath))
-
-    const result = await githubCreate(context,  figurePath, proposedName, content);
+    const figurePaths: { path: string, sha: string }[] = existingFigures.map((f: { path: string, sha: string; }) => ({ path: f.path, sha: f.sha }));
+    const figureToReplace = figurePaths.find(f => f.path === figurePath + '/' + name);
```

Comment:
`find(...)` is a neat function.

Reply:
Right? The only pitfall is that it only returns the first if many potential matches are present, but presumably we don't have that problem here.